Formula/rnp.rb

```diff
@@ -0,0 +1,37 @@
+class Rnp < Formula
+  desc "OpenPGP tools for encrypting, decrypting, signing, and verifying files"
+  homepage "https://github.com/rnpgp/rnp"
+  url "https://github.com/rnpgp/rnp.git", :tag => "v0.13.1"
+  head "https://github.com/rnpgp/rnp.git"
+
+  depends_on "cmake" => :build
+  depends_on "botan"
+  depends_on "json-c"
+
+  def install
+    botan = Formula["botan"]
+    jsonc = Formula["json-c"]
+
+    tag = `git describe`
+    ohai "Building tag #{tag}"
+    # only required to set when can't be determined automatically
+    # version tag
+
+    mkdir "build" do
+      system(
+        "cmake",
+        "-DBUILD_SHARED_LIBS=ON",
+        "-DBUILD_TESTING=OFF",
+        "-DCMAKE_INSTALL_PREFIX=#{prefix}",
+        "-DCMAKE_PREFIX_PATH=#{botan.prefix};#{jsonc.prefix}",
+        *std_cmake_args,
+        "..",
+      )
+      system "make", "install"
+    end
+  end
+
+  test do
+    system "rnp", "--version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.